pkg/resource/table/hooks.go

```diff
@@ -671,6 +695,14 @@ func customPreCompare(
 		}
 	}
 
+	if ackcompare.HasNilDifference(a.ko.Spec.ResourcePolicy, b.ko.Spec.ResourcePolicy) {
+		delta.Add("Spec.ResourcePolicy", a.ko.Spec.ResourcePolicy, b.ko.Spec.ResourcePolicy)
+	} else if a.ko.Spec.ResourcePolicy != nil && b.ko.Spec.ResourcePolicy != nil {
+		if *a.ko.Spec.ResourcePolicy != *b.ko.Spec.ResourcePolicy {
+			delta.Add("Spec.ResourcePolicy", a.ko.Spec.ResourcePolicy, b.ko.Spec.ResourcePolicy)
+		}
+	}
+
```

Comment:
Is this block the same the original generated code when we set `compare.is_ignore: false` ?

Reply:
Actually we do have a way to parse IAM policies that gets formatted/reformatted by the AWS APIs, checkout https://github.com/aws-controllers-k8s/iam-controller/blob/main/pkg/resource/role/hooks.go#L398-L432